src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -187,6 +180,24 @@ def register_options(cls, register):
 
   def register_extra_products_from_contexts(self, targets, compile_contexts):
     super(RscCompile, self).register_extra_products_from_contexts(targets, compile_contexts)
+
+    def cp_entry_for(maybe_file):
+      # Capture a snapshot for the file if it exists
+      if os.path.exists(maybe_file):
+        def capture_or_read_digest(scheduler, filename):
+          digest = load_digest(os.path.dirname(filename))
+          if not digest:
+            root = PathGlobsAndRoot(
+              PathGlobs(
+                (fast_relpath_optional(maybe_file, get_buildroot()),)),
+              text_type(get_buildroot()))
+            digest = scheduler.capture_snapshots((root,))[0].directory_digest
+          return digest
+
+        input_digest = capture_or_read_digest(self.context._scheduler, os.path.dirname(maybe_file))
+        return ClasspathEntry(maybe_file, input_digest)
+      else:
+        return maybe_file
```

Comment:
Hm. I think I'll have it return `None`. Skipping it is probably the best thing if the file doesn't exist yet, since we'll be filling in a new entry after the compile.